### ISSUE TITLE
A uikitview is below B uikitview, but A uikitview is refreshed, so that B uikitview cannot respond to the click event

### ISSUE THREAD

**Issue by chikang** [flutter/flutter#118366]
step：
1: Click the "Click Me" button, and a red B_uikitView native control pops up.
2: Then wait 5 seconds and the A_uikitView native control will show up,
3: At this point, the red B_uikitView native button is unresponsive

red  button dont click!!!!!!  why!!!!

my demo

[UIKitViewDemo.zip](https://github.com/flutter/flutter/files/10398076/UIKitViewDemo.zip)

**Comment by chikang** [comment]
![20230112-121526](https://user-images.githubusercontent.com/16316644/211974674-2745d133-57d3-4b88-a97f-a268789cdd43.gif)

**Comment by chikang** [comment]
problem ： When the red view comes out, the green button cannot be clicked

**Comment by exaby73** [comment]
I can reproduce this issue with the steps above. This may be a duplicate of https://github.com/flutter/flutter/issues/37579 but I cannot be certain of it. Labeling for further insights from the team

**Comment by chikang** [comment]
This doesn't look like  #37579，My problem is inevitable, and a demo is provided，so，how to solve this problem？

**Comment by chikang** [comment]
Is there no reply？

**Comment by exaby73** [comment]
@chikang The team has yet to look at this issue and prioritize it. Unfortunately, the only thing to do is wait for that to happen

**Comment by chikang** [comment]
> @chikang The team has yet to look at this issue and prioritize it. Unfortunately, the only thing to do is wait for that to happen

Well, I hope to hear from you soon  @exaby73

**Comment by hellohuanlin** [comment]
Did some initial investigation. From the debug view it looks like the green box is on top of the button: 

![Screenshot 2023-01-17 at 1 02 24 PM](https://user-images.githubusercontent.com/41930132/213011955-383e58a5-04c5-4a64-8540-26742ca68dad.png)

However, in the actual screen, the red button is not covered by the green box. So my guess is that flutter's platform view actually crop out the central area of green box so that it doesn't cover the red button. 

![simulator_screenshot_34614D99-00AB-4302-9225-259C16E724D1](https://user-images.githubusercontent.com/41930132/213012386-e33ed090-fb59-4543-bd1c-ec9981d87460.png)

@cyanglaz probably has more insight?

**Comment by cyanglaz** [comment]
Is the green uikitview intended to be below the red uikitview? I found the design a little odd.

**Comment by hellohuanlin** [comment]
@cyanglaz I think so, the red button is displayed using "showDialog" (which i assume should be the highest z order?): 

```
FloatingActionButton(
    onPressed: () {
      showDialog(
          context: context,
          builder: (context) {
            return Container(
              width: 300,
              height: 300,
              child: const UiKitView(
                viewType: 'B_uikitView',
                creationParamsCodec: StandardMessageCodec(),
              ),
            );
          });
    },
    child: Text(
      '点我',
      style: TextStyle(color: Colors.white, fontSize: 13),
    )),
],
```

**Comment by cyanglaz** [comment]
So it might be a issue with zIndex of PlatformView when displaying in a dialog? 

I think the click event worked as expected. The green view suppose to cover the red view, so the red view should not be handle touch events. The issue is the green view is displaying at the wrong zIndex.

**Comment by chikang** [comment]
> 因此，在对话盒中显示时，PlatformView 的 zIndex 可能存在问题？
> 
> 我认为点击事件按预期工作。绿色图像应该覆盖红色图像，所以红色图像不应该处理触摸事件。问题是绿色图像在图像中。

I also think it has something to do with zIndex, but how to solve it, I don't know

**Comment by chikang** [comment]
> So it might be a issue with zIndex of PlatformView when displaying in a dialog?
> 
> I think the click event worked as expected. The green view suppose to cover the red view, so the red view should not be handle touch events. The issue is the green view is displaying at the wrong zIndex.

I also think it has something to do with zIndex, but how to solve it, I don't know

**Comment by chikang** [comment]
How can red view get click events？

**Comment by chikang** [comment]
Is there a widget or API to solve this problem？

**Comment by KenwayDev** [comment]
![img_v2_d8ff81b8-43a3-4cc2-8011-b4583bef3fag](https://user-images.githubusercontent.com/106849500/215741143-9506fb99-1560-473e-8794-7b65ae23963f.png)

I had the same problem:
The view in flutter is presented as half screen black at the top of the screen, and the view is a WKWebview. But as you can see from the Xcode view hierarchy, half a screen of WKWebview is blocked by another UIView. The event in WKWebview is now unclickable。

@exaby73

**Comment by chikang** [comment]
Is there any progress?

**Comment by hellohuanlin** [comment]
> Is there any progress?

Hi, I got some good leads on this issue, but later got pulled into another issue so I paused it. Will resume after that issue is resolved.

**Comment by chikang** [comment]
Ok, I hope you can solve this problem.

**Comment by hellohuanlin** [comment]
@chikang yep, from where i left off, it should be solvable if im not mistaken.

**Comment by chikang** [comment]
If this problem is solved, will there be a beta branch

**Comment by jmagman** [comment]
@chikang please refer to https://github.com/flutter/flutter/wiki/Issue-hygiene#do-not-add-me-too-or-same-or-is-there-an-update-comments-to-bugs.  When there are update to report on this issue, it will be commented on and/or closed.
